### PR TITLE
Inlined the msmq create/open code to allows us to dispose properly

### PR DESCRIPTION
**Pull request by andreasohlund**
@ramonsmits I inlined the code to check for existence/create queues so that we can use `using` properly.

What do you think?

Targets #4304

### CODE REVIEW

File: src/NServiceBus.Core/Transports/Msmq/QueueCreator.cs [hunk 1]
```diff
@@ -39,29 +39,38 @@ void CreateQueueIfNecessary(string address, string identity)
...
             Logger.Debug($"Creating '{address}' if needed.");
 
-            MessageQueue queue = null;
+            if (msmqAddress.IsRemote)
+            {
+                Logger.Info($"'{address}' is a remote queue and won't be created");
+                return;
+            }
+
+
```
Comment (timbussmann) [inline src/NServiceBus.Core/Transports/Msmq/QueueCreator.cs]: remove extra line

File: src/NServiceBus.Core/Transports/Msmq/QueueCreator.cs [hunk 2]
```diff
@@ -39,29 +39,38 @@ void CreateQueueIfNecessary(string address, string identity)
...
+            catch (MessageQueueException ex)
             {
-                queue?.Dispose();
+                if (ex.MessageQueueErrorCode == MessageQueueErrorCode.QueueExists)
+                {
+                    //Solve the race condition problem when multiple endpoints try to create same queue (e.g. error queue).
+                    return;
+                }
+
+                throw;
```
Comment (timbussmann) [inline src/NServiceBus.Core/Transports/Msmq/QueueCreator.cs]: the rethrow seems to differ from the previous behavior, I assume this is intentionally?

**Comment by timbussmann** [comment]
are these changes really responsible for that failing test or what's now causing this?

File: src/NServiceBus.Core/Transports/Msmq/QueueCreator.cs (hunk 2 above)
Comment (andreasohlund) [inline src/NServiceBus.Core/Transports/Msmq/QueueCreator.cs]: Now that we check for remote queue before creating it doesn't seem to make any sense to swallow exceptions other than queue already exists?

**Comment by andreasohlund** [comment]
>are these changes really responsible for that failing test or what's now causing this?

Very likely, I'll debug!

File: src/NServiceBus.Core/Transports/Msmq/QueueCreator.cs (hunk 2 above)
Comment (andreasohlund) [inline src/NServiceBus.Core/Transports/Msmq/QueueCreator.cs]: Ie https://github.com/Particular/NServiceBus/pull/4310/files/0e56755d266ecc7d144e2ecf3507080a7d474517#diff-987fda24ac81af0d4ded1ccfae952e30R42

**Comment by andreasohlund** [comment]
@ramonsmits added unit tests https://github.com/Particular/NServiceBus/pull/4310/commits/66958aee44639872e539ee7b46018351ecc18106

I'll add a few more tomorrow

**Comment by andreasohlund** [comment]
Think I've found the issue. Since the old code swallowed all persmissions issues

https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Transports/Msmq/QueueCreator.cs#L55

We never detected that this test generates to long names for permissions to be set. I'll verify and fix

**Comment by andreasohlund** [comment]
@timbussmann confirmed, it was a to long name.

**Pull request merged**